packages/wrangler/src/cfetch/index.ts

```diff
@@ -162,6 +162,10 @@ function throwFetchError(
 	resource: string,
 	response: FetchResult<unknown>
 ): never {
+	// This is an error from within an MSW handler
+	if (typeof vitest !== "undefined" && !("errors" in response)) {
+		throw response;
+	}
```

Comment:
This is admittedly a horrible hack. We should move away from the pattern of using `expect()` within MSW handler, which will remove the need for this.